lpd/utils/threshold_checker.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC, abstractmethod
+from typing import Union
+
+from torch import Tensor
+
+
+class ThresholdChecker(ABC):
+    """
+    Check if the current value is better than the previous best value according to different threshold criteria
+    This is an abstract class meant to be inherited by different threshold checkers
+    Can also be inherited by the user to create custom threshold checkers
+    """
+    def __init__(self, threshold: float):
+        self.threshold = threshold
+
+    @abstractmethod
+    def is_new_value_higher(self, new_value: Union[float, Tensor], old_value: Union[float, Tensor]) -> bool:
```

Comment:
is_new_value_higher & is_new_value_lower functions are not "structured",
meaning, the main goal of lpd is that you know how everything behaves by 
reading the trainer.py

Look at CallbackMonitor for example, you define its behaviour in the ctor, and know
how it behaves.

i think we should align with it by defining the ThresholdChecker with the supported directions (higher/lower)
maybe create a new enum ? 
ThresholdCheckerMode.Higher
ThresholdCheckerMode.Lower

that way the ThresholdChecker will also only implement the default __call__ function
to check if condition is met or not